src/VirtoCommerce.Platform.Web/Swagger/CustomSwaggerGenerator.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.Swagger;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using VirtoCommerce.Platform.Core.Swagger;
+
+namespace VirtoCommerce.Platform.Web.Swagger
+{
+    /// <summary>
+    /// Generate swagger schema ids for schema refs depending on document name
+    /// </summary>
+    public class CustomSwaggerGenerator : ISwaggerProvider
+    {
+        private readonly SwaggerGenerator _swaggerGenerator;
+        private string _documentName;
+
+        public CustomSwaggerGenerator(SwaggerGeneratorOptions options, IApiDescriptionGroupCollectionProvider apiDescriptionsProvider, ISchemaGenerator schemaGenerator)
+        {
+            // We change schema generator options to replace SchemaIdSelector function to a document-dependent implementation
+            // Unfortunately this member declared as private, there is no another way to obtain options in this member.
+            var optionsFieldInfo = schemaGenerator.GetType().GetField("_generatorOptions", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            var schemaGeneratorOptions = (SchemaGeneratorOptions)optionsFieldInfo.GetValue(schemaGenerator);
```

Comment:
Can we register custom schema Id globally with using this extension?  `SchemaGeneratorOptions.CustomSchemaIds`